### PR TITLE
Relation created event

### PR DESCRIPTION
**Pull request by jameinel**
Built from #218 

I believe Dmitrii had already addressed all the original review points. This just adds support in the Harness for triggering relation-created when you call `.add_relation`.

A lot of the seeming churn is because Dmitrii had cleaned up a bunch of the doc strings as part of the previous review. The core change is just introducing the class and using it in the automatic define_event blocks.

### CODE REVIEW

File: ops/charm.py [hunk 1]
```diff
@@ -297,86 +245,71 @@ def restore(self, snapshot):
...
 
 
+class RelationCreatedEvent(RelationEvent):
+    """Represents the `relation-created` hook from Juju.
+
+    A relation-created event fires after a relation is created in Juju. A relation-joined
+    event will only fire after the respective relation-created event.
+    For units that are coming up, relation-created events fire after the install event
+    before any other events that fire after install, including leader-elected. This allows
+    a leader unit to work with peer application relation data early in its life cycle.
```
Comment (jameinel) [inline ops/charm.py:255]: We should also confirm that this reads clearly, I think it was the last bit that Gustavo was concerned about.

Comment (chipaca) [reply ops/charm.py:255]: It isn't clear in and of itself. Perhaps reiterating the order of events would help, as that is where it gets confusing. Or we could drop the whole thing after the first sentence for now and add the ordering elsewhere; it does feel like there needs to be a centralised "this is the order of things" somewhere, not spread out amongst the events.

Comment (facundobatista) [reply ops/charm.py:255]: Shouldn't this be explicited in some *Juju* docs? And we can just point to that.

**Comment by jameinel** [comment]
I tweaked it a bit, less to give the exact relation of it to all other charms, but hopefully enough to still give you the idea of how 'relation-created' differs from 'relation-joined' and 'relation-changed'.

**Pull request merged**